otx/algorithms/detection/adapters/mmdet/hooks/det_class_probability_map_hook.py

```diff
@@ -121,3 +124,123 @@ def forward_single(x, cls_convs, conv_cls):
                     "YOLOXHead, ATSSHead, SSDHead, VFNetHead."
                 )
         return cls_scores
+
+
+class MaskRCNNHook(BaseRecordingForwardHook):
```

Comment:
* How about changing the name of this class? `MaskRCNNRecordingForwardHook`? `MaskRCNNHook` itself is not proper naming for only the saliency map. 

* We could set the additional parent class for the detection task (couldn't we use the `DetClassProbabilityMapHook`?) for further implementations if we have the plan to implement other hooks such as `ATSSRecordingForwardHook`. And in this case, we also move `normalize` function to parent class for further use.

Reply:
I agree that `MaskRCNNRecordingForwardHook` might be more concreate. Updated

Regarding the OOP refactoring .
`DetClassProbabilityMapHook` is used for all single-stage detectors.
`normalize` is pretty general for all task types, so it can be moved to the upper-most level - `BaseRecordingForwardHook`. But I propose to do it in a separate PR, after @GalyaZalesskaya will also finish with sal map merge for tiling.
Except normalization, I do not see common code between `DetClassProbabilityMapHook` and `MaskRCNNRecordingForwardHook`, so probbaly for now base class for detectors is not needed.

Reply:
Thanks for kind reply

Reply:
I moved `normalize` parameter to `BaseRecordingForwardHook` in my [PR#2240](https://github.com/openvinotoolkit/training_extensions/pull/2240) related to XAI for tiling. After merging the current PR I'll rebase my PR onto this, and finish refactoring of saliency map hooks